src/bsoncxx/test/bson_value.cpp

```diff
@@ -38,139 +41,289 @@ void value_construction_test(bson_value::view test_view) {
 
     REQUIRE(test_value == test_view);
 }
-}  // namespace
 
-TEST_CASE("types::bson_value::value", "[bsoncxx::types::bson_value::value]") {
-    auto doc_value = make_document(kvp("hello", "world"));
-    auto doc2_value = make_document(kvp("a", 1));
+template <typename T, typename U>
+void coverting_construction_test(T actual, U expected) {
+    bson_value::value copy_initialization = actual;
+    REQUIRE(copy_initialization == expected);
 
-    auto doc = doc_value.view();
-    auto doc2 = doc2_value.view();
+    bson_value::value direct_initialization(actual);
+    REQUIRE(direct_initialization == expected);
 
-    auto elem = doc["hello"];
-    auto elem2 = doc2["a"];
+    bson_value::value _static_cast = static_cast<bson_value::value>(actual);
+    REQUIRE(_static_cast == expected);
+}
 
-    SECTION("can be constructed from a bson_value::view") {
+}  // namespace
+
+TEST_CASE("types::bson_value::value", "[bsoncxx::types::bson_value::value]") {
+    SECTION("can be constructed from") {
         SECTION("bool") {
-            auto test_doc = bson_value::make_value(types::b_bool{true});
+            auto test_doc = bson_value::make_value(b_bool{true});
             value_construction_test(test_doc.view());
+
+            coverting_construction_test(true, test_doc);
+            coverting_construction_test(b_bool{true}, test_doc);
         }
 
         SECTION("utf8") {
-            auto test_doc = bson_value::make_value("super duper");
+            std::string value = "super duper";
+
+            auto test_doc = bson_value::make_value(value);
             value_construction_test(test_doc.view());
 
+            coverting_construction_test(value, test_doc);
+            coverting_construction_test(value.c_str(), test_doc);
+            coverting_construction_test(b_utf8{value}, test_doc);
+            coverting_construction_test(stdx::string_view{value}, test_doc);
```

Comment:
I think this does need to be wrapped in a string view to pass the explicit length. Otherwise, the conversion will assume the string is null terminated to determine the length. I believe this is the cause of test failures on Windows: https://evergreen.mongodb.com/task_log_raw/cxx_driver_integration__os~windows_2k8_mongodb_version~4.0_compile_and_test_with_shared_libs_patch_11a863102c8549f82698ebe710394a0d3dd10c8a_5f90884332f417101242e124_20_10_21_19_13_08/0?type=T#L7212